### PR TITLE
Add gpt recipe

### PR DESCRIPTION
**Pull request by stuhlmueller**
### Brief summary of what the package does

This package defines a set of functions and variables for running instruction-following language models like GPT-3.  It allows the user to enter a command with history and completion, and optionally use the current region as input.  The output of the command is displayed in a temporary buffer with the same major mode as the original buffer.  The output is streamed as it is produced by the GPT process.  The user can view and export the command history to a file.

### Direct link to the package repository

https://github.com/stuhlmueller/gpt.el

### Your association with the package

Author and maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

**Comment by riscy** [comment]
Thanks for this.  Checklist looks great! 

Even though the .py file is not an elisp file, it's a good idea to add the standard license boilerplate to it as well.  flake8 also noted that the f-string on line 56 doesn't have any variables to interpolate.

I may be in the minority but I feel like this exception doesn't need to be caught and turned into a sys.exit --
```python
except ImportError:
    print("Error: OpenAI python package not installed.")
    print("Run pip install openai.")
    sys.exit(1)
```
e.g. why not just leave the ImportError intact? (`pip install openai` might not actually solve the problem e.g. if pip is pointing to another python installation, for instance.)

**Comment by stuhlmueller** [comment]
Thanks for the review!

> Even though the .py file is not an elisp file, it's a good idea to add the standard license boilerplate to it as well. flake8 also noted that the f-string on line 56 doesn't have any variables to interpolate.

Both done!

> I may be in the minority but I feel like this exception doesn't need to be caught and turned into a sys.exit

My thinking is that if someone installs the package from melpa without first seeing the README and we don't catch the `ImportError`, it will be unclear whether this is a problem with the package (should it have installed the dependency?) or expected. I [clarified](https://github.com/stuhlmueller/gpt.el/commit/1db27e69745b414b19407fdd7f01c55e66583e28) the error message slightly.

File: recipes/gpt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(gpt :fetcher github
+     :repo "stuhlmueller/gpt.el"
+     :files ("gpt.el" "gpt.py"))
```
Comment (riscy) [inline recipes/gpt:3]:
```suggestion
     :files (:defaults "gpt.py"))
```

**Review by riscy — COMMENTED** [review]
Looks good now - this is an easy merge, but I thought I would lastly suggest you add `:defaults` to your recipe.  This will give your repository room to grow within the bounds of the defaults (e.g. to include `gpt-test.el` files or whichever).

**Comment by stuhlmueller** [comment]
Good idea, added.

**Comment by riscy** [comment]
Thank you!  And welcome to MELPA. :)

**Pull request merged**